tests/5.1/target/test_target_memcpy_rect_async_depobj.c

```diff
@@ -0,0 +1,92 @@
+//===--- test_target_memcpy_rect_async_depobj.c ----------------------------===//
+//
+//  Inspired from OpenMP 5.1 Examples Doc, 5.16.4 & 8.9
+//  This test utilizes the omp_target_memcpy_rect_async construct to
+//  allocate memory on the device asynchronously. The construct
+//  uses 'obj' for dependency, so that memory is only copied once
+//  the variable listed in the depend clause is changed.
+//
+////===----------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+#include <math.h>
+
+#define N 5
+
+int errors, i, j;
+
+int test_target_memcpy_async_depobj() {
+
+    int h, t;
+    errors = 0;
+    double *devRect;
+    h = omp_get_initial_device();
+    t = omp_get_default_device();
+
+    double* hostRect[N];
+    for(i = 0; i < N; i++){
+        hostRect[i] = (double *)malloc( sizeof(double)*N); 
+    } // 5x5 2D array
+    devRect = (double *)omp_target_alloc( sizeof(double)*N, t);
```

Comment:
Due to the change above:
```suggestion
    devRect = (double **)omp_target_alloc( sizeof(double)*N, t);
```